client/src/app/pages/applications/applications-table/applications-table.tsx

```diff
@@ -194,6 +206,37 @@ export const ApplicationsTable: React.FC = () => {
     });
   };
 
+  const handleDownload = async () => {
+    const ids = selectedRows
+      .map((row) => row.tasks.currentAnalyzer?.id)
+      .filter((id): id is number => typeof id === "number");
+
+    try {
+      const tasks = await getTasksByIds(ids, selectedFormat);
+      const data =
+        selectedFormat === "yaml"
+          ? yaml.dump(tasks, { indent: 2 })
+          : JSON.stringify(tasks, null, 2);
```

Comment:
If the data is coming back in the requested "selectedFormat" why is it being formatted again?